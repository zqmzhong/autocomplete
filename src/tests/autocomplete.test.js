import { shallow, mount } from 'enzyme';
import AutoComplete from '../index';
import data from '../mocks/data.json';

describe('AutoComplete Test', () => {
    it('test render', () => {
        const componentWrapper = shallow(<AutoComplete data={[]} />);
        expect(componentWrapper.html()).toMatchSnapshot();
    });

    it('test render with clear icon', () => {
        const componentWrapper = shallow(<AutoComplete
            placeholder="input here"
            data={[]}
            allowClear={true}
        />);
        expect(componentWrapper.html()).toMatchSnapshot();
    });

    it('test mount', () => {
        const componentWrapper = mount(<AutoComplete
            placeholder="input here"
            data={[]}
            allowClear={true}
        />);
        expect(componentWrapper.find('input').length).toEqual(1);
    });

    it('test input text', () => {
        const spySearch = jest.fn();
        const componentWrapper = mount(<AutoComplete
            placeholder="input here"
            data={data}
            allowClear={true}
            onSearch={spySearch}
        />);
        const mockEvent = {
            target: { value: "test" }
        };
        componentWrapper.find('input').simulate('change', mockEvent);
        setTimeout(() => {
            expect(spySearch).toBeCalled();
        }, 500);
    });

    it('test select a item', () => {
        const spySelect = jest.fn();
        const componentWrapper = mount(<AutoComplete
            placeholder="input here"
            data={data}
            allowClear={true}
            onSelect={spySelect}
        />);
        setTimeout(() => {
            componentWrapper.find('button').simulate('click');
            expect(spySelect).toBeCalled();
        }, 500);
    });
});
