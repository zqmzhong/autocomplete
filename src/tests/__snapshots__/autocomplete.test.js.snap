// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete Test test render 1`] = `"<div class=\\"sc-bdfBwQ gWPNlw\\"><div style=\\"display:block;width:200vw;height:200vh;background-color:transparent;position:fixed;z-index:0;top:0;left:0\\"></div><div><input type=\\"text\\" id=\\"autocomplete-input\\" autoComplete=\\"off\\" value=\\"\\" placeholder=\\"\\" class=\\"sc-gsTCUz sc-gKsewC fBmtzO fQcUuh\\"/><span class=\\"sc-dlfnbm lkQlkV\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 16 16\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill-rule=\\"evenodd\\" d=\\"M11.854 4.146a.5.5 0 010 .708l-7 7a.5.5 0 01-.708-.708l7-7a.5.5 0 01.708 0z\\" clip-rule=\\"evenodd\\"></path><path fill-rule=\\"evenodd\\" d=\\"M4.146 4.146a.5.5 0 000 .708l7 7a.5.5 0 00.708-.708l-7-7a.5.5 0 00-.708 0z\\" clip-rule=\\"evenodd\\"></path></svg></span></div></div>"`;

exports[`AutoComplete Test test render with clear icon 1`] = `"<div class=\\"sc-bdfBwQ gWPNlw\\"><div style=\\"display:block;width:200vw;height:200vh;background-color:transparent;position:fixed;z-index:0;top:0;left:0\\"></div><div><input type=\\"text\\" id=\\"autocomplete-input\\" autoComplete=\\"off\\" value=\\"\\" placeholder=\\"input here\\" class=\\"sc-gsTCUz sc-gKsewC fBmtzO fQcUuh\\"/><span class=\\"sc-dlfnbm fYUQzM\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 16 16\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill-rule=\\"evenodd\\" d=\\"M11.854 4.146a.5.5 0 010 .708l-7 7a.5.5 0 01-.708-.708l7-7a.5.5 0 01.708 0z\\" clip-rule=\\"evenodd\\"></path><path fill-rule=\\"evenodd\\" d=\\"M4.146 4.146a.5.5 0 000 .708l7 7a.5.5 0 00.708-.708l-7-7a.5.5 0 00-.708 0z\\" clip-rule=\\"evenodd\\"></path></svg></span></div></div>"`;
